### ISSUE TITLE
flutter 'gen-l10n'" terminated with exit code: 1

### ISSUE THREAD

**Issue by sgehrman** [lsaudon/l10nization#374]
I'm getting this error, but running $ flutter gen-l10n in the terminal works fine.

I do have an unusual setup. I have a sub package that has the localization code.

In my main app I have a l10n.yaml file with 

```
arb-dir: localizations/lib/arb       // localizations is my sub package
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
synthetic-package: false
output-dir: localizations/lib/l10n    // localizations is my sub package
```

In the sub package, I have this: 

```
arb-dir: lib/arb
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
synthetic-package: false
output-dir: lib/l10n
```

I have the generate true in both the main app and the sub package

*  Executing task: /home/steve/development/flutter/bin/flutter gen-l10n 

Attempted to generate localizations code without having the flutter: generate flag turned on.
Check pubspec.yaml and ensure that flutter: generate: true has been added and rebuild the project. Otherwise, the localizations source code will not be importable.

 *  The terminal process "/home/steve/development/flutter/bin/flutter 'gen-l10n'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by sgehrman** [comment]
```
flutter doctor -v
[✓] Flutter (Channel stable, 3.16.8, on Debian GNU/Linux 12 (bookworm) 6.1.0-17-amd64, locale en_US.UTF-8)
    • Flutter version 3.16.8 on channel stable at /home/steve/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 67457e669f (8 days ago), 2024-01-16 16:22:29 -0800
    • Engine revision 6e2ea58a5c
    • Dart version 3.2.5
    • DevTools version 2.28.5

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /home/steve/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /home/steve/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • Debian clang version 14.0.6
    • cmake version 3.25.1
    • ninja version 1.11.1
    • pkg-config version 1.8.1

[✓] Android Studio (version 2023.1)
    • Android Studio at /home/steve/android-studio
    • Flutter plugin version 77.1.2
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.85.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.80.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Debian GNU/Linux 12 (bookworm) 6.1.0-17-amd64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 120.0.6099.224

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```
It's a flutter package.

I'm wondering what directory you are using when you issue that command?  the root?  maybe that's the issue?

**Comment by sgehrman** [comment]
OK, I think it must be the directory where it runs flutter gen-l10n.   It works if I do it in the main app, but doesn't work if I do it in another sub package.  I guess I would have to add this l10n.yaml to all my sub packages, but I would like to keep it simple if possible.  I have this large mono repo with a bunch of sub packages.

**Comment by sgehrman** [comment]
now I'm seeing another issue.

```
The 'arb-dir' directory, 'LocalDirectory: '/home/steve/Documents/GitHub/dfc/dashboard/shared_ui/localizations/lib/arb'', does not exist.
Make sure that the correct path was provided.
```

In another sub package I put in the l10n.yaml

```
arb-dir: localizations/lib/arb
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
synthetic-package: false
output-dir: localizations/lib/l10n
```

and that works when adding the string to the arb, but when it runs the gen-l10n it fails.  

the path should be ../localizations, but if I do that it fails when I add the string to the arb.

**Comment by lsaudon** [comment]
To test, I didn't know you could put translations in another package.😀

**Comment by sgehrman** [comment]
I have a complex app.  A desktop app that serves a web app and there are a bunch of sub packages that share code between them.

I get "no arb files found" when I do the ../localizations

**Comment by sgehrman** [comment]
going to sleep, will check tomorrow and respond if needed...

**Comment by lsaudon** [comment]
I'll try to create a similar project to see if I reproduce the bug.
It's not the same time at my place. 😉

**Comment by lsaudon** [comment]
Your l10n.yaml file in your application suggests that your folder looks like this.
```
.
├── l10n.yaml
├── lib
│   └── main.dart
├── localizations
│   ├── l10n.yaml
│   ├── lib
│   │   ├── l10n
│   │   │   ├── arb
│   │   │   │   ├── app_en.arb
│   │   │   │   └── app_es.arb
│   │   │   └── l10n.dart
│   │   └── localizations.dart
│   └── pubspec.yaml
├── pubspec.lock
└── pubspec.yaml
```
If it's not this, your arb-dir and output-dir is wrong.

**Comment by sgehrman** [comment]
It's lib/arb, but I checked those values and it looks correct.

**Comment by lsaudon** [comment]
You don't have a other package with a l10n.yaml in your project ?

**Comment by sgehrman** [comment]
I have another sub package, that's the one that doesn't work.  The app works, and the localization package works, but the other sub package has to have ..localizations in the path, and it seems that it doesn't like the ..

app:  localizations/xxx/arb
localizations: xxx/arb
subpackage: ../localizations/xxx/arb (doesn't work right)

Another little thing I noticed which would be nice to fix.  

'This is John\'s stuff'  => "This is John\\'s stuff"
but
"This is John's stuff" => "This is John's stuff"

It would be good if you could not do the \\ and just use the '

Thanks.

**Comment by sgehrman** [comment]
To summarize:

This works with flutter gen-l10n
but your code says it can't find the arb files

```
arb-dir: ../localizations/lib/arb
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
synthetic-package: false
output-dir: ../localizations/lib/l10n
```

This works in your code, but flutter gen-l10n fails
```
arb-dir: localizations/lib/arb
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
synthetic-package: false
output-dir: localizations/lib/l10n
```

**Comment by lsaudon** [comment]
Have you tried running with VSCode `Command Palette > Flutter: Generate Localizations` ?

**Comment by sgehrman** [comment]
Just tried it, same problem.

**Comment by lsaudon** [comment]
If you get the same error when running the command manually. The bug must come from the [flutter extension](https://github.com/Dart-Code/Dart-Code).

**Comment by sgehrman** [comment]
The bug is in your plugin.

If I correctly set ../localiztions, flutter gen-l10n works, but your plugin doesn't.  It says 'can't find arb file'.

**Comment by lsaudon** [comment]
Alright, I found the bug.
I parse the path incorrectly and `path/../localizations/arb/` doesn't work.

**Comment by lsaudon** [comment]
I've published a new version `3.1.0`.
Let me know if it fixes the bug.

**Comment by sgehrman** [comment]
Thanks! Seems to work now. But I'll test it more later.  I'm basically done with my localization now, but I think I have a few strings left.
